app/src/main/java/com/libs/facerecognition/FacePreprocessor.java

```diff
@@ -100,7 +100,7 @@ public void waitForTask(Task<List<Face>> task) {
      * @param face  Face from face detection
      * @return rotated image
      */
-    private Bitmap rotateAndTrimFace(Bitmap image, Face face) {
+    public Bitmap rotateAndTrimFace(Bitmap image, Face face) {
```

Comment:
I don't think if changing function encapsultion just to test it is good idea. On the other hand, private functions generally should not be tested explicility, as their functionality should be validated while testing public API of the class. In our case I think we shouldn't test the rotateAndTrimFace and isFaceOnImage methods, if they are used in the public API use case. 
If you really need to to that though, I will recommend using reflection instead of changing the code just for the test case.

Reply:
i agree, but this method can be public. Maybe in future we want to test somewhere else if face is on image or not.